tests/core/client.spec.ts

```diff
@@ -0,0 +1,152 @@
+import * as elliptic from 'elliptic';
+import * as puppeteer from 'puppeteer';
+import { BTCPayClient } from '../../src/core/client';
+import { Cryptography as myCrypto } from '../../src/core/cryptography';
+
+const USER_NAME = 'test@example.com';
+const PASSWORD = 'satoshinakamoto';
+const URL = 'https://testnet.demo.btcpayserver.org/';
+
+const MY_PRIVATE_KEY = Buffer.from(
+  '31eb31ecf1a640c9d1e0a1105501f36235f8c7d51d67dcf74ccc968d74cb6b25',
+  'hex',
+);
+
+const STORE_ID = 'HPPHFtqtsKsF3KU18fBNwVGP64hicGoRynvQrC3R2Rkw';
+const TOKENS = {
+  merchant: '96ukQNT5eoNwQyRjjpgbRMvbHa6iqAJ436Zu5gRVuWxf',
+};
+
+const INVOICE_ID = 'TRnwXeAkuLQihe22mJs7J4';
+
+const loginAndGetPairingCode = async (): Promise<any> => {
+  const newTokenName = 'autotest ' + new Date().getTime();
+
+  const browser = await puppeteer.launch({ headless: true });
```

Comment:
Couldn't get the tests to run successfully until I changed this to:
```
const browser = await puppeteer.launch({
  headless: true,
  args: ['--no-sandbox'],
});
```

Reply:
also, please rebase to the current `master`

Reply:
https://github.com/GoogleChrome/puppeteer/blob/master/docs/troubleshooting.md#setting-up-chrome-linux-sandbox

Removing sandbox seems dangerous...

What OS/version are you on?

Reply:
I added an ENV variable that will tell people to use `BTCPAY_IGNORE_SANDBOX_ERROR=1`

People who have a sandbox should use it by default.

also, there is an entry about Travis setup, apparently Travis needs to install sandbox support to use puppeteer.

https://github.com/GoogleChrome/puppeteer/blob/master/docs/troubleshooting.md#running-puppeteer-on-travis-ci

docker alpine also has trouble apparently.

Reply:
> https://github.com/GoogleChrome/puppeteer/blob/master/docs/troubleshooting.md#setting-up-chrome-linux-sandbox
> 
> Removing sandbox seems dangerous...
> 
> What OS/version are you on?

I was using Arch Linux to do the testing. It appears that Debian based OSes also have the same problem too.

When I was trying to debug why the tests weren't running, I also noticed the issue with Travis as well. Might need to set that environment variable on Travis. Should probably make a note about that in the README?